### PR TITLE
Improving decoding error logging and handling

### PR DESCRIPTION
**Pull request by mvs5465**
Added more exception handling to better log errors in decoding files.

### CODE REVIEW

File: jpo-ode-plugins/src/main/java/us/dot/its/jpo/ode/plugin/j2735/oss/OssAsn1Coder.java [hunk 1]
```diff
@@ -95,28 +100,54 @@ public Asn1Object UPER_DecodeBsmStream(InputStream ins) {
...
+                logger.debug("DECODER - Message Size = {}", ins.available());
                 coder.decode(ins, bsm);
                 gbsm = OssBsm.genericBsm(bsm);
             }
-        } catch (Exception e) {
-            logger.debug("Error decoding ", e);
+        } catch (DecodeFailedException e) {
+            AbstractData partialDecodedMessage = e.getDecodedData();
+            if (partialDecodedMessage != null) {
+                logger.debug("DECODER - Warning, message only partially decoded: {}", partialDecodedMessage);
```
Comment (hmusavi) [inline jpo-ode-plugins/src/main/java/us/dot/its/jpo/ode/plugin/j2735/oss/OssAsn1Coder.java]: This one should be a logger.error. We do not expect to receive partial messages.

File: jpo-ode-plugins/src/main/java/us/dot/its/jpo/ode/plugin/j2735/oss/OssAsn1Coder.java [hunk 2]
```diff
@@ -95,28 +100,54 @@ public Asn1Object UPER_DecodeBsmStream(InputStream ins) {
...
+            } else {
+                logger.debug("DECODER - Warning, invalid bytes detected in file.");
```
Comment (hmusavi) [inline jpo-ode-plugins/src/main/java/us/dot/its/jpo/ode/plugin/j2735/oss/OssAsn1Coder.java]: I suggest this message: "Ignoring extraneous bytes detected at the end of the input stream."

File: jpo-ode-plugins/src/main/java/us/dot/its/jpo/ode/plugin/j2735/oss/OssAsn1Coder.java [hunk 3]
```diff
@@ -95,28 +100,54 @@ public Asn1Object UPER_DecodeBsmStream(InputStream ins) {
...
-    
+
     @Override
     public Asn1Object UPER_DecodeMessageFrameStream(InputStream ins) {
         MessageFrame mf = new MessageFrame();
         J2735MessageFrame gmf = null;
 
         try {
             if (ins.available() > 0) {
+                logger.debug("DECODER - Message Size = {}", ins.available());
```
Comment (hmusavi) [inline jpo-ode-plugins/src/main/java/us/dot/its/jpo/ode/plugin/j2735/oss/OssAsn1Coder.java]: This is going to generate a lot of logs. Please comment it out.

File: jpo-ode-plugins/src/main/java/us/dot/its/jpo/ode/plugin/j2735/oss/OssAsn1Coder.java (hunk 1 above)
Comment (mvs5465) [inline jpo-ode-plugins/src/main/java/us/dot/its/jpo/ode/plugin/j2735/oss/OssAsn1Coder.java]: Made this change.

File: jpo-ode-plugins/src/main/java/us/dot/its/jpo/ode/plugin/j2735/oss/OssAsn1Coder.java (hunk 2 above)
Comment (mvs5465) [inline jpo-ode-plugins/src/main/java/us/dot/its/jpo/ode/plugin/j2735/oss/OssAsn1Coder.java]: Made this change.

File: jpo-ode-plugins/src/main/java/us/dot/its/jpo/ode/plugin/j2735/oss/OssAsn1Coder.java (hunk 3 above)
Comment (mvs5465) [inline jpo-ode-plugins/src/main/java/us/dot/its/jpo/ode/plugin/j2735/oss/OssAsn1Coder.java]: Made this change.

File: jpo-ode-plugins/src/main/java/us/dot/its/jpo/ode/plugin/j2735/oss/OssAsn1Coder.java (hunk 1 above)
Comment (hmusavi) [inline jpo-ode-plugins/src/main/java/us/dot/its/jpo/ode/plugin/j2735/oss/OssAsn1Coder.java]: did you? Still shows as debug not error.

File: jpo-ode-plugins/src/main/java/us/dot/its/jpo/ode/plugin/j2735/oss/OssAsn1Coder.java [hunk 4]
```diff
@@ -115,8 +131,21 @@ public Asn1Object UPER_DecodeMessageFrameStream(InputStream ins) {
                 coder.decode(ins, mf);
                 gmf = OssMessageFrame.genericMessageFrame(mf);
             }
-        } catch (Exception e) {
-            logger.debug("Error decoding ", e);
+        } catch (DecodeFailedException e) {
+            AbstractData partialDecodedMessage = e.getDecodedData();
+            if (partialDecodedMessage != null) {
+                logger.debug("DECODER - Warning, message only partially decoded: {}", partialDecodedMessage.toString());
```
Comment (hmusavi) [inline jpo-ode-plugins/src/main/java/us/dot/its/jpo/ode/plugin/j2735/oss/OssAsn1Coder.java]: logger.error?

File: jpo-ode-plugins/src/main/java/us/dot/its/jpo/ode/plugin/j2735/oss/OssAsn1Coder.java (hunk 1 above)
Comment (mvs5465) [inline jpo-ode-plugins/src/main/java/us/dot/its/jpo/ode/plugin/j2735/oss/OssAsn1Coder.java]: Woops, didn't push it. Should be changed now.

**Pull request merged**